File: Capítulo_3/ECMAScript 6 e as novas funcionalidades de array/Ordenação personalizada/Código em Python/compareFunction.py
friends = [
    { "name": "John", "age": 30 },
    { "name": "Ana", "age": 20 },
    { "name": "Chris", "age": 25 }
]

def get_age(person):
    return person["age"]

# Em Python para se utilizar a função de comparação em vez de sort usa-se sorted contudo este método so aceita apenas um argumento
print(sorted(friends, key = get_age))

File: Capítulo_3/ECMAScript 6 e as novas funcionalidades de array/Ordenação personalizada/compareFunction.js
/* 
 Podemos ordenar um array que contenha qualquer tipo de objeto  e podemos também criar uma compareFunction para comparar os elementos de acordo com o que for necessário. Por exemplo, suponha que haja um objeto Person com name e age e queremos ordenar o array de acordo com a idade (age) da pessoa. Podemos usar o código a seguir:
*/
const friends = [
    { name: "John", age: 30 },
    { name: "Ana", age: 20 },
    { name: "Chris", age: 25 }, // vírgula no final: ES2017
];

function comparePerson(a, b) {
    if (a.age < b.age) {
        return -1;
    } else if (a.age > b.age) {
        return 1;
    } else {
        return 0;
    }
}
console.log(friends.sort(comparePerson))
